src/main/java/org/newrelic/nrjmx/v2/JMXFetcher.java

```diff
@@ -291,12 +296,14 @@ private List<String> getMBeanAttributeNames(ObjectName objectName) throws JMXCon
             if (internalStat != null) {
                 internalStat.setSuccessful(true);
             }
-        } catch (ConnectException ce) {
+        } catch (JMXConnectionError je) {
+            throw je;
+        } catch (IOException io) {
```

Comment:
from documentation:
Throws:
IOException – A communication problem occurred when talking to the MBean server.